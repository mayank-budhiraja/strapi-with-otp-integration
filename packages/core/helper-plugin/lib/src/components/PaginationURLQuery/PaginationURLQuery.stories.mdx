<!--- PaginationURLQuery.stories.mdx --->

import { useEffect } from 'react';
import { Meta, ArgsTable, Canvas, Story } from '@storybook/addon-docs';
import { Main, Box } from '@strapi/design-system';
import useQueryParams from '../../hooks/useQueryParams';
import PaginationURLQuery from './index';

<Meta title="components/PaginationURLQuery" />

# PaginationURLQuery

This component is used to change the page size.

## Imports

```js
import { PaginationURLQuery } from '@strapi/helper-plugin';
```

## Usage

<Canvas>
  <Story name="base">
    {() => {
      const [{ query }, setQuery] = useQueryParams();
      useEffect(() => {
        setQuery({ pageSize: 10, page: 1, sort: 'firstname' });
      }, []);
      return (
        <Main>
          <Box paddingTop={6}>
            <PaginationURLQuery pagination={{ pageCount: 10 }} />
          </Box>
        </Main>
      );
    }}
  </Story>
</Canvas>

### Props

<ArgsTable of={PaginationURLQuery} />
