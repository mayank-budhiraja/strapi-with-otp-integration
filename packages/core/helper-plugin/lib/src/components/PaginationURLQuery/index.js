/**
 * Pagination
 *
 * The component works as follows
 * `1` , 2, 3, ... 10
 * 1, `2`, 3, ... 10
 * 1, 2, `3`, 4, ... 10
 * 1, 2, 3, `4`, 5, ... 10
 * 1, ..,4, `5`, 6, ... 10
 *
 * 1, ...., 8, 9, `10`
 * 1, ...., 8, `9`, 10
 * 1, ...., 7, `8`, 9, 10
 * 1, ... 6, `7`, 8, 9, 10
 */

import React from 'react';
import {
  NextLink,
  Pagination,
  PreviousLink,
  Dots,
  PageLink,
} from '@strapi/design-system/Pagination';
import PropTypes from 'prop-types';
import { useLocation, NavLink } from 'react-router-dom';
import { useIntl } from 'react-intl';
import { stringify } from 'qs';
import useQueryParams from '../../hooks/useQueryParams';

const PaginationURLQuery = ({ pagination: { pageCount } }) => {
  const [{ query }] = useQueryParams();
  const activePage = parseInt(query?.page || '1', 10);
  const { pathname } = useLocation();
  const { formatMessage } = useIntl();
  const makeSearch = page => stringify({ ...query, page }, { encode: false });
  const nextSearch = makeSearch(activePage + (pageCount > 1 ? 1 : 0));

  const previousSearch = makeSearch(activePage - 1);

  const firstLinks = [
    <PageLink as={NavLink} key={1} number={1} to={`${pathname}?${makeSearch(1)}`}>
      {formatMessage(
        { id: 'components.pagination.go-to', defaultMessage: 'Go to page {page}' },
        { page: 1 }
      )}
    </PageLink>,
  ];

  if (pageCount <= 4) {
    const links = Array.from({ length: pageCount })
      .map((_, i) => i + 1)
      .map(number => {
        return (
          <PageLink
            as={NavLink}
            key={number}
            number={number}
            to={`${pathname}?${makeSearch(number)}`}
          >
            {formatMessage(
              { id: 'components.pagination.go-to', defaultMessage: 'Go to page {page}' },
              { page: number }
            )}
          </PageLink>
        );
      });

    return (
      <Pagination activePage={activePage} pageCount={pageCount}>
        <PreviousLink as={NavLink} to={`${pathname}?${previousSearch}`}>
          {formatMessage({
            id: 'components.pagination.go-to-previous',
            defaultMessage: 'Go to previous page',
          })}
        </PreviousLink>
        {links}
        <NextLink as={NavLink} to={`${pathname}?${nextSearch}`}>
          {formatMessage({
            id: 'components.pagination.go-to-next',
            defaultMessage: 'Go to next page',
          })}
        </NextLink>
      </Pagination>
    );
  }

  let firstLinksToCreate = [];
  let lastLinks = [];
  let lastLinksToCreate = [];
  const middleLinks = [];

  if (pageCount > 1) {
    lastLinks.push(
      <PageLink
        as={NavLink}
        key={pageCount}
        number={pageCount}
        to={`${pathname}?${makeSearch(pageCount)}`}
      >
        {formatMessage(
          { id: 'components.pagination.go-to', defaultMessage: 'Go to page {page}' },
          { page: pageCount }
        )}
      </PageLink>
    );
  }

  if (activePage === 1 && pageCount >= 3) {
    firstLinksToCreate = [2];
  }

  if (activePage === 2 && pageCount >= 3) {
    if (pageCount === 5) {
      firstLinksToCreate = [2, 3, 4];
    } else if (pageCount === 3) {
      firstLinksToCreate = [2];
    } else {
      firstLinksToCreate = [2, 3];
    }
  }

  if (activePage === 4 && pageCount >= 3) {
    firstLinksToCreate = [2];
  }

  if (activePage === pageCount && pageCount >= 3) {
    lastLinksToCreate = [pageCount - 1];
  }

  if (activePage === pageCount - 2 && pageCount > 3) {
    lastLinksToCreate = [activePage + 1, activePage, activePage - 1];
  }

  if (activePage === pageCount - 3 && pageCount > 3 && activePage > 5) {
    lastLinksToCreate = [activePage + 2, activePage + 1, activePage, activePage - 1];
  }

  if (activePage === pageCount - 1 && pageCount > 3) {
    lastLinksToCreate = [activePage, activePage - 1];
  }

  lastLinksToCreate.forEach(number => {
    lastLinks.unshift(
      <PageLink as={NavLink} key={number} number={number} to={`${pathname}?${makeSearch(number)}`}>
        Go to page {number}
      </PageLink>
    );
  });

  firstLinksToCreate.forEach(number => {
    firstLinks.push(
      <PageLink as={NavLink} key={number} number={number} to={`${pathname}?${makeSearch(number)}`}>
        {formatMessage(
          { id: 'components.pagination.go-to', defaultMessage: 'Go to page {page}' },
          { page: number }
        )}
      </PageLink>
    );
  });

  if (
    ![1, 2].includes(activePage) &&
    activePage <= pageCount - 3 &&
    firstLinks.length + lastLinks.length < 6
  ) {
    const middleLinksToCreate = [activePage - 1, activePage, activePage + 1];

    middleLinksToCreate.forEach(number => {
      middleLinks.push(
        <PageLink
          as={NavLink}
          key={number}
          number={number}
          to={`${pathname}?${makeSearch(number)}`}
        >
          {formatMessage(
            { id: 'components.pagination.go-to', defaultMessage: 'Go to page {page}' },
            { page: number }
          )}
        </PageLink>
      );
    });
  }

  const shouldShowDotsAfterFirstLink =
    pageCount > 5 || (pageCount === 5 && (activePage === 1 || activePage === 5));
  const shouldShowMiddleDots = middleLinks.length > 2 && activePage > 4 && pageCount > 5;

  const beforeDotsLinksLength = shouldShowMiddleDots
    ? pageCount - activePage - 1
    : pageCount - firstLinks.length - lastLinks.length;
  const afterDotsLength = shouldShowMiddleDots
    ? pageCount - firstLinks.length - lastLinks.length
    : pageCount - activePage - 1;

  return (
    <Pagination activePage={activePage} pageCount={pageCount}>
      <PreviousLink as={NavLink} to={`${pathname}?${previousSearch}`}>
        {formatMessage({
          id: 'components.pagination.go-to-previous',
          defaultMessage: 'Go to previous page',
        })}
      </PreviousLink>
      {firstLinks}
      {shouldShowMiddleDots && (
        <Dots>
          {formatMessage(
            {
              id: 'components.pagination.remaining-links',
              defaultMessage: 'And {number} other links',
            },
            { number: beforeDotsLinksLength }
          )}
        </Dots>
      )}
      {middleLinks}
      {shouldShowDotsAfterFirstLink && (
        <Dots>
          {formatMessage(
            {
              id: 'components.pagination.remaining-links',
              defaultMessage: 'And {number} other links',
            },
            { number: afterDotsLength }
          )}
        </Dots>
      )}
      {lastLinks}
      <NextLink as={NavLink} to={`${pathname}?${nextSearch}`}>
        {formatMessage({
          id: 'components.pagination.go-to-next',
          defaultMessage: 'Go to next page',
        })}
      </NextLink>
    </Pagination>
  );
};

PaginationURLQuery.propTypes = {
  pagination: PropTypes.shape({ pageCount: PropTypes.number.isRequired }).isRequired,
};

export default PaginationURLQuery;
