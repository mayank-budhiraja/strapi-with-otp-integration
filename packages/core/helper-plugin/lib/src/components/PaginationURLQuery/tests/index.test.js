/**
 *
 * Tests for PaginationURLQuery
 *
 */

import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { ThemeProvider, lightTheme } from '@strapi/design-system';
import PaginationURLQuery from '../index';

const messages = {
  'components.pagination.go-to': 'Go to page {page}',
  'components.pagination.go-to-previous': 'Go to previous page',
  'components.pagination.remaining-links': 'And {number} other links',
  'components.pagination.go-to-next': 'Go to next page',
};

const makeApp = (history, pageCount) => (
  <Router history={history}>
    <ThemeProvider theme={lightTheme}>
      <IntlProvider locale="en" messages={messages} textComponent="span">
        <PaginationURLQuery pagination={{ pageCount }} />
      </IntlProvider>
    </ThemeProvider>
  </Router>
);

describe('<PaginationURLQuery />', () => {
  it('renders and matches the snapshot', () => {
    const history = createMemoryHistory();
    const {
      container: { firstChild },
    } = render(makeApp(history, 1));

    expect(firstChild).toMatchInlineSnapshot(`
      .c0 {
        -webkit-align-items: center;
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        -webkit-flex-direction: row;
        -ms-flex-direction: row;
        flex-direction: row;
      }

      .c1 > * + * {
        margin-left: 4px;
      }

      .c4 {
        border: 0;
        -webkit-clip: rect(0 0 0 0);
        clip: rect(0 0 0 0);
        height: 1px;
        margin: -1px;
        overflow: hidden;
        padding: 0;
        position: absolute;
        width: 1px;
      }

      .c7 {
        font-weight: 600;
        color: #32324d;
        font-size: 0.75rem;
        line-height: 1.33;
      }

      .c8 {
        line-height: revert;
      }

      .c2 {
        padding: 12px;
        border-radius: 4px;
        -webkit-text-decoration: none;
        text-decoration: none;
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        position: relative;
        outline: none;
      }

      .c2:after {
        -webkit-transition-property: all;
        transition-property: all;
        -webkit-transition-duration: 0.2s;
        transition-duration: 0.2s;
        border-radius: 8px;
        content: '';
        position: absolute;
        top: -4px;
        bottom: -4px;
        left: -4px;
        right: -4px;
        border: 2px solid transparent;
      }

      .c2:focus-visible {
        outline: none;
      }

      .c2:focus-visible:after {
        border-radius: 8px;
        content: '';
        position: absolute;
        top: -5px;
        bottom: -5px;
        left: -5px;
        right: -5px;
        border: 2px solid #4945ff;
      }

      .c5 {
        padding: 12px;
        border-radius: 4px;
        box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
        -webkit-text-decoration: none;
        text-decoration: none;
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        position: relative;
        outline: none;
      }

      .c5:after {
        -webkit-transition-property: all;
        transition-property: all;
        -webkit-transition-duration: 0.2s;
        transition-duration: 0.2s;
        border-radius: 8px;
        content: '';
        position: absolute;
        top: -4px;
        bottom: -4px;
        left: -4px;
        right: -4px;
        border: 2px solid transparent;
      }

      .c5:focus-visible {
        outline: none;
      }

      .c5:focus-visible:after {
        border-radius: 8px;
        content: '';
        position: absolute;
        top: -5px;
        bottom: -5px;
        left: -5px;
        right: -5px;
        border: 2px solid #4945ff;
      }

      .c6 {
        color: #271fe0;
        background: #ffffff;
      }

      .c6:hover {
        box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
      }

      .c3 {
        font-size: 0.7rem;
        pointer-events: none;
      }

      .c3 svg path {
        fill: #c0c0cf;
      }

      .c3:focus svg path,
      .c3:hover svg path {
        fill: #c0c0cf;
      }

      <nav
        aria-label="pagination"
        class="sc-eBTqsU"
      >
        <ul
          class="c0 c1"
        >
          <li>
            <a
              aria-current="page"
              aria-disabled="true"
              class="c2 c3 active"
              href="/"
              tabindex="-1"
            >
              <div
                class="c4"
              >
                Go to previous page
              </div>
              <svg
                aria-hidden="true"
                fill="none"
                height="1em"
                viewBox="0 0 10 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.88 14.12L3.773 8 9.88 1.88 8 0 0 8l8 8 1.88-1.88z"
                  fill="#32324D"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="c5 c6 active"
              href="/?page=1"
            >
              <div
                class="c4"
              >
                Go to page 1
              </div>
              <span
                aria-hidden="true"
                class="c7 c8"
              >
                1
              </span>
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              aria-disabled="true"
              class="c2 c3 active"
              href="/"
              tabindex="-1"
            >
              <div
                class="c4"
              >
                Go to next page
              </div>
              <svg
                aria-hidden="true"
                fill="none"
                height="1em"
                viewBox="0 0 10 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 1.88L6.107 8 0 14.12 1.88 16l8-8-8-8L0 1.88z"
                  fill="#32324D"
                />
              </svg>
            </a>
          </li>
        </ul>
      </nav>
    `);
  });

  it('should display 3 links when the active page is 1 and the pageCount is greater than 5', () => {
    const history = createMemoryHistory();
    render(makeApp(history, 10));

    expect(screen.getByText('Go to page 1')).toBeInTheDocument();
    expect(screen.getByText('Go to page 2')).toBeInTheDocument();
    expect(screen.queryByText('Go to page 3')).not.toBeInTheDocument();
    expect(screen.getByText('And 8 other links')).toBeInTheDocument();
    expect(screen.getByText('Go to page 10')).toBeInTheDocument();
  });

  it('should change the page correctly', async () => {
    const history = createMemoryHistory();
    history.push({ pathname: '/test', search: 'page=3&pageSize=10&sort=firstname' });
    render(makeApp(history, 10));

    expect(screen.getByText('Go to page 1')).toBeInTheDocument();
    expect(screen.getByText('Go to page 2')).toBeInTheDocument();
    expect(screen.queryByText('Go to page 3')).toBeInTheDocument();
    expect(screen.queryByText('Go to page 4')).toBeInTheDocument();
    expect(screen.queryByText('Go to page 5')).not.toBeInTheDocument();
    expect(screen.getByText('And 6 other links')).toBeInTheDocument();
    expect(screen.getByText('Go to page 10')).toBeInTheDocument();

    fireEvent.click(screen.getByText('Go to page 4'));

    await waitFor(() => {
      expect(history.location.pathname).toEqual('/test');
      expect(history.location.search).toEqual('?page=4&pageSize=10&sort=firstname');
      expect(screen.queryByText('Go to page 5')).toBeInTheDocument();
    });
  });

  it('should display the dots correctly', () => {
    const history = createMemoryHistory();
    history.push({ pathname: '/test', search: 'page=5&pageSize=10&sort=firstname' });
    const {
      container: { firstChild },
    } = render(makeApp(history, 10));

    expect(firstChild).toMatchSnapshot();
  });

  it('should work when the pageCount is inferior or equal to 4', async () => {
    const history = createMemoryHistory();
    history.push({ pathname: '/test', search: 'page=1&pageSize=10&sort=firstname' });
    const {
      container: { firstChild },
    } = render(makeApp(history, 4));

    expect(firstChild).toMatchSnapshot();

    fireEvent.click(screen.getByText('Go to page 2'));

    await waitFor(() => {
      expect(history.location.pathname).toEqual('/test');
      expect(history.location.search).toEqual('?page=2&pageSize=10&sort=firstname');
      expect(screen.getByText('Go to page 1')).toBeInTheDocument();
      expect(screen.getByText('Go to page 2')).toBeInTheDocument();
      expect(screen.queryByText('Go to page 3')).toBeInTheDocument();
      expect(screen.queryByText('Go to page 4')).toBeInTheDocument();
    });

    fireEvent.click(screen.getByText('Go to page 3'));

    await waitFor(() => {
      expect(history.location.pathname).toEqual('/test');
      expect(history.location.search).toEqual('?page=3&pageSize=10&sort=firstname');
      expect(screen.getByText('Go to page 1')).toBeInTheDocument();
      expect(screen.getByText('Go to page 2')).toBeInTheDocument();
      expect(screen.queryByText('Go to page 3')).toBeInTheDocument();
      expect(screen.queryByText('Go to page 4')).toBeInTheDocument();
    });

    fireEvent.click(screen.getByText('Go to page 4'));

    await waitFor(() => {
      expect(history.location.pathname).toEqual('/test');
      expect(history.location.search).toEqual('?page=4&pageSize=10&sort=firstname');
      expect(screen.getByText('Go to page 1')).toBeInTheDocument();
      expect(screen.getByText('Go to page 2')).toBeInTheDocument();
      expect(screen.queryByText('Go to page 3')).toBeInTheDocument();
      expect(screen.queryByText('Go to page 4')).toBeInTheDocument();
    });
  });
});
