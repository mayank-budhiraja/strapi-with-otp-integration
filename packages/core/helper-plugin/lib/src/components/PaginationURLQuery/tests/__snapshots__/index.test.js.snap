// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationURLQuery /> should display the dots correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 > * + * {
  margin-left: 4px;
}

.c4 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c6 {
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c11 {
  font-weight: 600;
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c7 {
  line-height: revert;
}

.c2 {
  padding: 12px;
  border-radius: 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  outline: none;
}

.c2:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c2:focus-visible {
  outline: none;
}

.c2:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c9 {
  padding: 12px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  outline: none;
}

.c9:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c9:focus-visible {
  outline: none;
}

.c9:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c5 {
  color: #32324d;
}

.c5:hover {
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c10 {
  color: #271fe0;
  background: #ffffff;
}

.c10:hover {
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c3 {
  font-size: 0.7rem;
}

.c3 svg path {
  fill: #666687;
}

.c3:focus svg path,
.c3:hover svg path {
  fill: #4a4a6a;
}

.c8 {
  color: #32324d;
}

<nav
  aria-label="pagination"
  class="sc-eBTqsU"
>
  <ul
    class="c0 c1"
  >
    <li>
      <a
        aria-current="page"
        aria-disabled="false"
        class="c2 c3 active"
        href="/test?page=4&pageSize=10&sort=firstname"
      >
        <div
          class="c4"
        >
          Go to previous page
        </div>
        <svg
          aria-hidden="true"
          fill="none"
          height="1em"
          viewBox="0 0 10 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.88 14.12L3.773 8 9.88 1.88 8 0 0 8l8 8 1.88-1.88z"
            fill="#32324D"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        class="c2 c5 active"
        href="/test?page=1&pageSize=10&sort=firstname"
      >
        <div
          class="c4"
        >
          Go to page 1
        </div>
        <span
          aria-hidden="true"
          class="c6 c7"
        >
          1
        </span>
      </a>
    </li>
    <li>
      <div
        class="c2 c8"
      >
        <div
          class="c4"
        >
          And 4 other links
        </div>
        <span
          aria-hidden="true"
          class="c6 c7"
        >
          …
        </span>
      </div>
    </li>
    <li>
      <a
        aria-current="page"
        class="c2 c5 active"
        href="/test?page=4&pageSize=10&sort=firstname"
      >
        <div
          class="c4"
        >
          Go to page 4
        </div>
        <span
          aria-hidden="true"
          class="c6 c7"
        >
          4
        </span>
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        class="c9 c10 active"
        href="/test?page=5&pageSize=10&sort=firstname"
      >
        <div
          class="c4"
        >
          Go to page 5
        </div>
        <span
          aria-hidden="true"
          class="c11 c7"
        >
          5
        </span>
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        class="c2 c5 active"
        href="/test?page=6&pageSize=10&sort=firstname"
      >
        <div
          class="c4"
        >
          Go to page 6
        </div>
        <span
          aria-hidden="true"
          class="c6 c7"
        >
          6
        </span>
      </a>
    </li>
    <li>
      <div
        class="c2 c8"
      >
        <div
          class="c4"
        >
          And 8 other links
        </div>
        <span
          aria-hidden="true"
          class="c6 c7"
        >
          …
        </span>
      </div>
    </li>
    <li>
      <a
        aria-current="page"
        class="c2 c5 active"
        href="/test?page=10&pageSize=10&sort=firstname"
      >
        <div
          class="c4"
        >
          Go to page 10
        </div>
        <span
          aria-hidden="true"
          class="c6 c7"
        >
          10
        </span>
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        aria-disabled="false"
        class="c2 c3 active"
        href="/test?page=6&pageSize=10&sort=firstname"
      >
        <div
          class="c4"
        >
          Go to next page
        </div>
        <svg
          aria-hidden="true"
          fill="none"
          height="1em"
          viewBox="0 0 10 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 1.88L6.107 8 0 14.12 1.88 16l8-8-8-8L0 1.88z"
            fill="#32324D"
          />
        </svg>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`<PaginationURLQuery /> should work when the pageCount is inferior or equal to 4 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 > * + * {
  margin-left: 4px;
}

.c4 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c7 {
  font-weight: 600;
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c10 {
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c8 {
  line-height: revert;
}

.c2 {
  padding: 12px;
  border-radius: 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  outline: none;
}

.c2:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c2:focus-visible {
  outline: none;
}

.c2:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c5 {
  padding: 12px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  outline: none;
}

.c5:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c5:focus-visible {
  outline: none;
}

.c5:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c6 {
  color: #271fe0;
  background: #ffffff;
}

.c6:hover {
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c9 {
  color: #32324d;
}

.c9:hover {
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c3 {
  font-size: 0.7rem;
  pointer-events: none;
}

.c3 svg path {
  fill: #c0c0cf;
}

.c3:focus svg path,
.c3:hover svg path {
  fill: #c0c0cf;
}

.c11 {
  font-size: 0.7rem;
}

.c11 svg path {
  fill: #666687;
}

.c11:focus svg path,
.c11:hover svg path {
  fill: #4a4a6a;
}

<nav
  aria-label="pagination"
  class="sc-eBTqsU"
>
  <ul
    class="c0 c1"
  >
    <li>
      <a
        aria-current="page"
        aria-disabled="true"
        class="c2 c3 active"
        href="/test"
        tabindex="-1"
      >
        <div
          class="c4"
        >
          Go to previous page
        </div>
        <svg
          aria-hidden="true"
          fill="none"
          height="1em"
          viewBox="0 0 10 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.88 14.12L3.773 8 9.88 1.88 8 0 0 8l8 8 1.88-1.88z"
            fill="#32324D"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        class="c5 c6 active"
        href="/test?page=1&pageSize=10&sort=firstname"
      >
        <div
          class="c4"
        >
          Go to page 1
        </div>
        <span
          aria-hidden="true"
          class="c7 c8"
        >
          1
        </span>
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        class="c2 c9 active"
        href="/test?page=2&pageSize=10&sort=firstname"
      >
        <div
          class="c4"
        >
          Go to page 2
        </div>
        <span
          aria-hidden="true"
          class="c10 c8"
        >
          2
        </span>
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        class="c2 c9 active"
        href="/test?page=3&pageSize=10&sort=firstname"
      >
        <div
          class="c4"
        >
          Go to page 3
        </div>
        <span
          aria-hidden="true"
          class="c10 c8"
        >
          3
        </span>
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        class="c2 c9 active"
        href="/test?page=4&pageSize=10&sort=firstname"
      >
        <div
          class="c4"
        >
          Go to page 4
        </div>
        <span
          aria-hidden="true"
          class="c10 c8"
        >
          4
        </span>
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        aria-disabled="false"
        class="c2 c11 active"
        href="/test?page=2&pageSize=10&sort=firstname"
      >
        <div
          class="c4"
        >
          Go to next page
        </div>
        <svg
          aria-hidden="true"
          fill="none"
          height="1em"
          viewBox="0 0 10 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 1.88L6.107 8 0 14.12 1.88 16l8-8-8-8L0 1.88z"
            fill="#32324D"
          />
        </svg>
      </a>
    </li>
  </ul>
</nav>
`;
